quickwit-metastore/src/metastore.rs

```diff
@@ -80,8 +80,9 @@ pub struct SplitMetadata {
     /// If a timestamp field is available, the min / max timestamp in the split.
     pub time_range: Option<RangeInclusive<i64>>,
 
-    /// Number of merges this segment has been subjected to during its lifetime.
-    pub generation: usize,
+    /// Encodes the type of demux operation the split has gone through.
+    /// Only splits who went through the same demux signature will be considered for demux.
+    pub demux_signature: String,
```

Comment:
Keep some meta information about previous operations sounds interesting. Of course we will have some logs but we can store at the split level operation count by operation type for example. Not super useful for now certainly.

Reply:
Removing demux_signature for the moment as it is premature.
It was not just a problem of observability. The idea was to use this information to identify which set of split can be demuxed together.